zerver/lib/markdown/help_relative_links.py

```diff
@@ -114,7 +138,11 @@ def run(self, lines: List[str]) -> List[str]:
         return lines
 
     def handleMatch(self, match: Match[str]) -> str:
-        return LINK_TYPE_HANDLERS[match.group("link_type")](match.group("key"))
+        link_type = match.group("link_type")
+        key = match.group("key")
+        if key is None:
+            key = link_type
+        return LINK_TYPE_HANDLERS[link_type](key)
```

Comment:
Could we not do `{relative|message|drafts}` and `{relative|message|scheduled}`? That way we could keep the current regex and handleMatch behavior? And the pattern for relative help links stays consistent in the help center documentation?

And then we have `message_handle_match`, which does the same thing you've got here, changing `overlay_info` to `message_info` or `message_type` maybe?

Reply:
Good point. I suggested this approach because we are also missing relative help links for `#recent_topics`/`#recent`, `#all_messages`, `#keyboard-shortcuts`, `#message-formatting`, `#search-operators`, `#invite`, and `#about-zulip`. I figured maybe we could handle all of these cases using a different function / pattern, but perhaps it would be best to do what you are suggesting for `#drafts` and `#scheduled`, and then we could add the following relative help links in a follow-up PR:

`{relative|message|recent}`
`{relative|message|all_messages}`

`{relative|gear|keyboard-shortcuts}`
`{relative|gear|message-formatting}`
`{relative|gear|search-operators}`
`{relative|gear|invite}`
`{relative|gear|about-zulip}`

What do you think?

Reply:
I think that's a good plan for follow-ups with those additions. I think the pattern you have here for `#drafts` and `#scheduled` would work for `#recent` and `#all_messages`. Do we need to do `#recent_topics` since this documentation would only be for server versions after that being changed to `#recent`?

From a git-grep, it does look like we have `{relative|gear|invite}` and `{relative|gear|about-zulip}` in use already. 

Though the invite one is broken because we don't use that hash in the web-app anymore, so we should fix that as a separate issue and replace those relative links with an include file with the shared instruction text.

```console
$ git grep '{relative|gear' help/
help/analytics.md:{relative|gear|stats}
help/include/review-organization-settings-instructions.md:{relative|gear|organization-settings}
help/invite-new-users.md:{relative|gear|invite}
help/invite-new-users.md:{relative|gear|invite}
help/review-your-organization-settings.md:{relative|gear|organization-settings}
help/review-your-settings.md:{relative|gear|settings}
help/review-your-settings.md:{relative|gear|settings}
help/review-your-settings.md:{relative|gear|settings}
help/view-zulip-version.md:{relative|gear|about-zulip}
```

Reply:
> Do we need to do `#recent_topics` since this documentation would only be for server versions after that being changed to `#recent`?
> 
> From a git-grep, it does look like we have `{relative|gear|invite}` and `{relative|gear|about-zulip}` in use already.

Oh you are right. We do not need to do `#recent_topics` or `#about-zulip`, and we should remove/replace `{relative|gear|invite}` with an include block.


> I think that's a good plan for follow-ups with those additions. I think the pattern you have here for `#drafts` and `#scheduled` would work for `#recent` and `#all_messages`. 

Okay so to sum up, this PR and the follow-up would add the following relative links:

`{relative|drafts}`
`{relative|scheduled}`
`{relative|recent}`
`{relative|all_messages}`

`{relative|gear|keyboard-shortcuts}`
`{relative|gear|message-formatting}`
`{relative|gear|search-operators}`

Reply:
Oh, I wasn't clear in my feedback, apologies. I think we should use `{relative|message|drafts}`, etc., and not change the current regex and handleMatch behavior.

I think how you've handled the default text for these situations ~ as a third entry in the object ~ works well since unlike the gear menu options, these instructions are going to be different for each situation (drafts, scheduled messages, recent conversations, etc.).

```python
message_info = {
    "drafts": ["Drafts", "/#drafts", draft_instructions],
    "scheduled": ["Scheduled messages", "/#scheduled", scheduled_instructions],
}
```

Reply:
Yeah, sounds good to me. Thanks for clarifying!

Reply:
I updated the PR and it should be good to go.